src/XML/src/FMI3/fmi3_xml_model_description.c

```diff
@@ -159,23 +157,15 @@ void fmi3_xml_clear_model_description( fmi3_xml_model_description_t* md) {
     /* --- variables --- */
     {
         /* free memory inside the variables */
-        /* NOTE: we have to use variablesByName here, because if parsing fails the other lists won't exist */
-        for (i = 0; i < jm_vector_get_size(jm_named_ptr)(&md->variablesByName); i++) {
-            fmi3_xml_variable_t* var = (fmi3_xml_variable_t*)jm_vector_get_item(jm_named_ptr)(&md->variablesByName, i).ptr;
-            fmi3_xml_variable_free_internals(md->callbacks, var);
-        }
-        /* free extra pointers */
-        if (md->variablesOrigOrder) {
-            jm_vector_free(jm_voidp)(md->variablesOrigOrder);
-            md->variablesOrigOrder = 0;
-        }
-        /* free extra pointers */
-        if (md->variablesByVR) {
-            jm_vector_free(jm_voidp)(md->variablesByVR);
-            md->variablesByVR = 0;
+        for (i = 0; i < jm_vector_get_size(jm_voidp)(&md->variablesOrigOrder); i++) {
```

Comment:
variablesOrigOrder is now the list owning the memory, so had to swap things around a bit since it's not a jm_named_ptr but a jm_voidp